Extend document tooltip with thumbnail preview.

phgross opened the pull request:
This PR reworks the look and feel of the document tooltip:
 - The tooltip includes now a thumbnail of the preview (closes #1977)
 - The complete Dokument link is now "tooltipable", not only the document icon.
 - The tooltip opens now on the right side of the pointer, this has the advantage that other document links for example in a listing are still visible.
 - Both delays (show/remove) of the tooltip were optimized.
 - Add icons to the actions.

![bildschirmfoto 2016-06-30 um 10 49 14](https://cloud.githubusercontent.com/assets/485755/16482519/56b714de-3eb0-11e6-9767-6d228d50a164.png)

Styling in a separate PR in plonetheme.teamraum

@lukasgraf @deiferni

deiferni commented on opengever/document/renderer.py:
```diff
@@ -40,44 +40,48 @@ def preview_link(self):
         label = translate(pmf(u'Edit metadata'), context=getRequest())
-        return '<a href="{}">{}</a>'.format(link, label.decode('utf-8'))
+        return '<a class="action-edit" href="{}">{}</a>'.format(
+            link, label.decode('utf-8'))
```
This looks fishy, should probably be `encode` since we're formatting a non-unicode string, or an unicode string.

deiferni commented on opengever/document/renderer.py:
```diff
@@ -40,44 +40,48 @@ def preview_link(self):
-            links.append(self.download_link())
+            links.append('<li>{}</li>'.format(self.download_link()))
 
         return """
```
`"\n".join(...)` ?

deiferni commented on opengever/document/renderer.py:
```diff
@@ -98,18 +102,30 @@ def get_showroom_data(self):
+    def get_bumblebee_thumbnail(self):
+        if self.is_bumblebeeable():
+            return """
+            <img src='{image_url}' alt='{title}' width="200" class="file-preview" />
```
`"\n<img..."` ?

deiferni commented on opengever/document/tests/test_renderer.py:
```diff
@@ -159,3 +160,35 @@ def test_download_link_is_only_available_for_documents(self, browser):
+    layer = OPENGEVER_FUNCTIONAL_BUMBLEBEE_LAYER
+
+    @browsing
+    def test_document_link_is_extend_with_showrom_data(self, browser):
```
extend**ed**

deiferni commented:
👍

phgross commented:
I've updated the PR. In addition to the thumbnail preview I optimized the implementation of the `DocumentLinkRender` as we disscussed earlier this week:
 - The Linkgeneration uses now a PageTemplate for the html generation
 - Renamed the `DocumentLinkRenderer` to `DocumentLinkWidget`
 - Displays the breadcrumb as a list

@lukasgraf or @deiferni please have another look in particular the last four commits.

deiferni commented:
Looks good to me :+1:

lukasgraf commented on opengever/document/widgets/document_link.pt:
```diff
@@ -0,0 +1,59 @@
+
+  <div class='tabbedview-tooltip-data'>
+    <div class='tooltip-header' tal:content="view/get_title" />
+    <ul class='tooltip-breadcrumb' tal:define="breadcrumbs context/get_breadcrumbs">
```
There's no `>` being used here, and I don't see any CSS that would style those list elements in such a way - are Breadcrumbs actually still separated with `>` ?

phgross replied:
See plonetheme.teamraum PR https://github.com/4teamwork/plonetheme.teamraum/pull/438/files#diff-dffe8f3324837116b7a3b77aa20f1587R35

lukasgraf commented:
Other than my comment and the failing test: Looks good to me too 👍

phgross commented:
@lukasgraf tests fixed ...

lukasgraf commented:
👍
